syntax = "proto3";
package query_service_ts;

service QueryServiceTs {
     rpc QueryBySchema(SchemaId) returns (TimeSeries);
     rpc QueryByRange(Range) returns (TimeSeries);
     rpc QueryRaw (RawStatement) returns (ValueBytes);
}

message Range {
    string object_id = 1;
    string start = 2;
    string end = 3;
    string step = 4;
}

message SchemaId {
    string schema_id = 1;
}

message TimeSeries {
    string timeseries = 1;
}

message RawStatement {
    string raw_statement = 1;
}

message ValueBytes {
    bytes value_bytes = 1;
}
